StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs

```diff
@@ -64,14 +64,15 @@ internal static bool IsLastInLine(this SyntaxToken token)
         /// <returns>true if token is preceded by a whitespace, otherwise false.</returns>
         internal static bool IsPrecededByWhitespace(this SyntaxToken token)
         {
-            SyntaxTriviaList triviaList = token.LeadingTrivia;
-            if (triviaList.Count > 0)
+            // Perf directly access the text instead of trivia
+            int pos = token.Span.Start - 1;
+
+            if (pos < 0)
             {
-                return triviaList.Last().IsKind(SyntaxKind.WhitespaceTrivia);
+                return false;
             }
 
-            triviaList = token.GetPreviousToken().TrailingTrivia;
-            return triviaList.Count > 0 && triviaList.Last().IsKind(SyntaxKind.WhitespaceTrivia);
+            return char.IsWhiteSpace(token.SyntaxTree.GetText()[pos]);
```

Comment:
:exclamation: `GetText(cancellationToken)`

:question: Should we handle the case where `SyntaxTree` is null (e.g. a token created by `SyntaxFactory.Token` that isn't in the tree yet)?